MerlinAU.sh

```diff
@@ -227,12 +232,18 @@ _DoExit_()
    _ReleaseLock_ ; exit "$exitCode"
 }
 
-##----------------------------------------------##
-## Added/Modified by Martinski W. [2023-Nov-18] ##
-##----------------------------------------------##
+##----------------------------------------##
+## Modified by Martinski W. [2024-May-31] ##
+##----------------------------------------##
+## To support new "3006" F/W Basecode ##
+if [ "$fwInstalledBaseVers" -eq 3006 ]
```

Comment:
I'm wondering if we should be hard coding these base versions as 3006, or if we should say anything higher than base version 3004?

Reply:
> I'm wondering if we should be hard coding these base versions as 3006, or if we should say anything higher than base version 3004?
>
Done in latest commit https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/226/commits/e49ae4a42dbd7cc0081ebeabd8c915f7aa96d52f

Reply:
I love how fast you are Martinski, it's like watching the super hero "the flash" work at coding hahaha

Reply:
> I love how fast you are Martinski, it's like watching the super hero "the flash" work at coding hahaha
>
I'll be testing the new code later this evening to make sure things are OK with the older AC & AX routers.
Hopefully, all our testing will pass with flying colors :>).

Reply:
I'll be testing this evening as well!

I'm currently out doing a bit of shopping but as soon as I'm back this is priority for me.

What's funny is I also thought about the upcoming 3006 releases, but as for as I went was just making sure we didn't hard code any 3004 values

Reply:
> I'll be testing this evening as well!
> 
> I'm currently out doing a bit of shopping but as soon as I'm back this is priority for me.
> 
> What's funny is I also thought about the upcoming 3006 releases, but as for as I went was just making sure we didn't hard code any 3004 values
>
I decided to make the initial changes after reviewing the changelog for the latest Alpha 3006.102.1 release and saw that they were all fairly easy to do. Hopefully, no major changes will be needed once the Beta or Production releases are out.

Reply:
> > I'll be testing this evening as well!
> > I'm currently out doing a bit of shopping but as soon as I'm back this is priority for me.
> > What's funny is I also thought about the upcoming 3006 releases, but as for as I went was just making sure we didn't hard code any 3004 values
> 
> I decided to make the initial changes after reviewing the changelog for the latest Alpha 3006.102.1 release and saw that they were all fairly easy to do. Hopefully, no major changes will be needed once the Beta or Production releases are out.

Yeah I think you hit most of the requirements here. You handled the change to the Kong version format so that a lower version of the newer firmware will properly be compared to the higher versions of the older firmware.

You handled the changelog changes which are obviously needed. I can't think of anything more off the top of my head.

Oh and you handled the LED nvram variables. My question is how do you know that's the new nvram variables? Did you look at the source code? Ask someone testing the new alphas? Or did you notice it in the changelogs?

I'm wondering if there's anymore of those types of changes we need to track down.

You also obviously handled a couple of bugs and improvements so naturally we should merge those in

Reply:
> > 
> > I decided to make the initial changes after reviewing the changelog for the latest Alpha 3006.102.1 release and saw that they were all fairly easy to do. Hopefully, no major changes will be needed once the Beta or Production releases are out.
> 
> Yeah I think you hit most of the requirements here. You handled the change to the long version format so that a lower version of the newer firmware will properly be compared to the higher versions of the older firmware.
> 
> You handled the changelog changes which are obviously needed. I can't think of anything more off the top of my head.
> 
> Oh and you handled the LED nvram variables. My question is how do you know that's the new nvram variables? Did you look at the source code? Ask someone testing the new alphas? Or did you notice it in the changelogs?
> 
>
Yeah, that info was provided in the Alpha version changelog.

>
> I'm wondering if there's anymore of those types of changes we need to track down.
> 
 I suppose there could be more, and hopefully, RMerlin will let us know in advance before the production release is out, especially if any changes were made to the built-in F/W Update process which our script relies heavily on.

>
> You also obviously handled a couple of bugs and improvements so naturally we should merge those in
>
Yeah, I'm now wondering if a new production release with those fixes will be needed soon since there are some scenarios where the "F/W Changelog Check Approval" does not quite work as expected.

Reply:
@Martinski4GitHub 

I say we just jump it out this weekend? Unless you wanted to wait for other changes?

Reply:
> @Martinski4GitHub
> 
> I say we just jump it out this weekend? Unless you wanted to wait for other changes?
>
At this point, I have no other changes to make. I'm waiting for the next Alpha or Beta 3006.102 release to review the changelog and see if anything else has been changed that would affect our script.